test/src/TestUtilities/SystemUtil.hpp

```diff
@@ -42,4 +43,31 @@ namespace Catch
     }
 }
 
+namespace sf::Testing
+{
+    class TemporaryFile
+    {
+    private:
+        std::string m_path;
+
+    public:
+        // Create a temporary file with a randomly generated path, containing 'contents'.
+        TemporaryFile(const std::string& contents);
+
+        // Close and delete the generated file.
+        ~TemporaryFile();
+
+        // Prevent copies.
+        TemporaryFile(const TemporaryFile&) = delete;
+        TemporaryFile& operator=(const TemporaryFile&) = delete;
+
+        // Allow moves.
+        TemporaryFile(TemporaryFile&&) = default;
+        TemporaryFile& operator=(TemporaryFile&&) = default;
```

Comment:
Shouldn't those move operations swap each `m_path` explicitly?

Reply:
That's not the real issue, the issue is that the destructor would be invoked for the moved-from object and misbehave. I have removed the move operations as they are not currently used in this PR.